packages/components/src/components/icon/icon.tsx

```diff
@@ -55,6 +56,7 @@ const Icon = React.forwardRef(
                 id={id}
                 width={width || size}
                 onClick={onClick}
+                onMouseDown={onMouseDown}
```

Comment:
I added onMouseDown as it wouldn't register the click when I am clicking on close circle icon, but instead it would just delete the entire payment method (delete icon)